src/main/java/pl/sda/climbing_shop/configuration/UserDetailsServiceImplForCustomers.java

```diff
@@ -0,0 +1,79 @@
+package pl.sda.climbing_shop.configuration;
+
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+import pl.sda.climbing_shop.address.Address;
+import pl.sda.climbing_shop.customer.Customer;
+import pl.sda.climbing_shop.customer.CustomerRepository;
+import pl.sda.climbing_shop.order.Order;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+@Component
+public class UserDetailsServiceImplForCustomers implements UserDetailsService {
+
+    private final CustomerRepository customerRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        Optional<Customer> optionalCustomer = this.customerRepository.findByEmail(username);
+        Customer customer = optionalCustomer.orElseThrow(() -> new UsernameNotFoundException("Customer not found"));
+
+        return ClimbingShopUserDetails.builder()
+                .authorities(List.of(customer.getRole().getName()).stream()
+                        .map(SimpleGrantedAuthority::new)
+                        .collect(Collectors.toList()))
+                .username(customer.getEmail())
+                .password(customer.getHashPassword())
+                .accountNonExpired(true)
+                .accountNonLocked(true)
+                .credentialsNonExpired(true)
+                .enabled(true)
```

Comment:
czy te rzeczy nie powinny być ciągniete z bazy? czy moze to nie przewidzane w pierwszej fazie  ? ;P

Reply:
dobra, ogarnelam, że spirngowy inteface userdetails wymaga ;)